### DISCUSSION TITLE
How to login using SSH key?

### DISCUSSION THREAD

**Discussion by anta40** [veeso/termscp#209] [Q&A] [answered]
I login to my development VPS using SSH key.
This is how I setup my connection:

```
Protocol: SCP
Remote host: xxx.xxx.xxx.xxx
Port number: 22
Username: anta40
Password: ....

```
OK, I'm supposed to use the correct SSH key, not the password. How to do it? On other SCP clients, I can easily pick it.
By pressing Ctrl+C (to show the configuration parameters), this is what I see under "SSH configuration path": `~/.ssh/config` (which is correct). And `anta40` is the same username in `~/.ssh/config`.

termscp always gives me this:
`authentication failed ([Session(18)] Authentication failed (username/password))`

Perhaps I miss something?
I'm on macOS, btw.

**Comment by veeso** [comment, marked as answer]
The ssh configuration file should work, if it doesn't there's something wrong with it or it has patterns that are not supported by the library.

Anyway, there is built-in storage which always works.
https://github.com/veeso/termscp/blob/main/docs/man.md#ssh-key-storage-

**Reply by anta40** [reply]
Ah I see. For me, the UI is a bit un-intuitive that you have to open another menu to provide the SSH key (on other SCP clients, when you are setting up a new connection, usually the SSH key field is located under the password).
Got it, anyway.
